### PR TITLE
Fix source amplitude in spectrum and source-time plotting

### PR DESCRIPTION
**Pull request by weiliangjin2021**
Reported in https://flow360.atlassian.net/browse/SCEM-2316

Ideally, I think we should update `amp_time`, but it will require changes in backend as well. So I simply fix it inside the plot function.

### CODE REVIEW

**Review by tylerflex — APPROVED** [review]
This seems fine to me for now, but ideally we should put the amplitude into the `self.amp_time()` directly? Let's make a note to change that (+backend fix) at some point?

**Comment by momchil-flex** [comment]
No complications, should be an easy backend fix. Well, of course I'll only be sure of it once we run the tests. :)

**Review by tylerflex — APPROVED** [review]

**Comment by tylerflex** [comment]
Looks good, I'll let you merge when the backend changes are in or whenever you think it's ready. Thanks!

**Comment by momchil-flex** [comment]
Ok yeah so there was the subtlety that the user-defined amplitude needs to be taken out of the normalization spectrum. Note that this should not change anything, i.e. the normalization is still the same - it's just that previously, `source_time.amplitude` was missing from the source spectrum. Now that it's there, we have to take it out in the normalization, which makes sense, since I want the `amplitude` and `phase` I provide to have an effect on the results, I just don't want the spurious part of the source spectrum to be there.

Have a look @tylerflex @shashwat-sh @weiliangjin2021

**Comment by shashwat-sh** [comment]
Makes sense to me. Do the backend tests pass now?

**Comment by momchil-flex** [comment]
Yes.

File: tidy3d/components/data/sim_data.py [hunk 1]
```diff
@@ -154,8 +154,11 @@ def source_spectrum_fn(freqs):
             """Source amplitude as function of frequency."""
             spectrum = source_time.spectrum(times, freqs, dt, complex_fields)
 
-            # remove user defined phase from normalization so its effect is present in the result
-            return spectrum * np.conj(user_defined_phase)
+            # Remove user defined amplitude and phase from the normalization
+            # such that they would still have an effect on the output fields.
+            # In other words, we are only normalizing out the arbitrary part of the spectrum
+            # that depends on things like freq0, fwidth and offset.
+            return spectrum / source_time.amplitude / user_defined_phase
```
Comment (tylerflex) [inline tidy3d/components/data/sim_data.py:161]: why is the user defined phase divided out here instead of multiplied after np.conj()?

Comment (momchil-flex) [reply tidy3d/components/data/sim_data.py:161]: It's equivalent, I just thought it makes it clear what we're doing (basically removing a user-defined complex number out of the spectrum).

It seems that `user_defined_phase` is actually not used anywhere else, so I could just make that line 

```
return spectrum / source_time.amplitude / np.exp(1j * source_time.phase)
```

Comment (momchil-flex) [reply tidy3d/components/data/sim_data.py]: Note: updated the code as above.

**Pull request merged**